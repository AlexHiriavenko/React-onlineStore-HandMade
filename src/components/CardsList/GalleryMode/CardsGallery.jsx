import Card from "./Card.jsx";
import s from "./cardsGallery.module.scss";

function CardsGallery(props) {
    const { btnAssignment, btnText, cards } = props;

    return (
        <section className={s.cards} data-testid="gallery">
            {cards.map((card) => (
                <Card
                    key={card.articul}
                    card={card}
                    btnAssignment={btnAssignment}
                    btnText={btnText}
                />
            ))}
        </section>
    );
}

export default CardsGallery;
