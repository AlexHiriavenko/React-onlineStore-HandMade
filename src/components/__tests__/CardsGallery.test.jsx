import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import CardsGallery from "./../CardsList/GalleryMode/CardsGallery";

const initialState = {
    cartCardsReducer: {
        cartCards: [],
    },
    favoritCardsReducer: {
        favoritCards: [],
    },
};

const mockStore = configureStore();
const store = mockStore(initialState);

test("renders CardsGallery component with cards", () => {
    const cards = [
        { articul: 1, title: "Card 1" },
        { articul: 2, title: "Card 2" },
        { articul: 3, title: "Card 3" },
    ];
    const btnAssignment = "modalAddToCart";
    const btnText = "Add to Cart";

    render(
        <Provider store={store}>
            <CardsGallery cards={cards} btnAssignment={btnAssignment} btnText={btnText} />
        </Provider>
    );

    const cardElements = screen.getAllByTestId("card");
    expect(cardElements).toHaveLength(cards.length);
    expect(screen.getByTestId("gallery")).toMatchSnapshot();
});
