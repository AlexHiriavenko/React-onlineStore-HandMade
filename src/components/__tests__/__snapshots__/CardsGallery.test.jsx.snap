// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CardsGallery component with cards 1`] = `
<section
  class="cards"
  data-testid="gallery"
>
  <div
    class="card"
    data-cards="true"
    data-testid="card"
    id="1"
  >
    <h2
      class="text title"
    />
    <img
      width="280"
    />
    <div
      class="cardContent"
    >
      <div>
        <p
          class="text"
        >
          <span
            class="prop"
          >
            Price:
          </span>
        </p>
        <p
          class="text"
        >
          <span
            class="prop"
          >
            Articul:
          </span>
          1
        </p>
        <p
          class="text"
        >
          <span
            class="prop"
          >
            Color:
          </span>
        </p>
      </div>
      <div
        class="actions"
      >
        <button
          class="btn"
          data-assignment="modalAddToCart"
          style="background-color: brown; color: bisque;"
        >
          Add to Cart
        </button>
        <div
          class="wrapStar"
        >
          <svg
            height="28"
            style="cursor: pointer;"
            viewBox="0 0 51 48"
            width="28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              Five Pointed Star
            </title>
            <path
              d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
              fill="bisque"
              stroke=""
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <div
    class="card"
    data-cards="true"
    data-testid="card"
    id="2"
  >
    <h2
      class="text title"
    />
    <img
      width="280"
    />
    <div
      class="cardContent"
    >
      <div>
        <p
          class="text"
        >
          <span
            class="prop"
          >
            Price:
          </span>
        </p>
        <p
          class="text"
        >
          <span
            class="prop"
          >
            Articul:
          </span>
          2
        </p>
        <p
          class="text"
        >
          <span
            class="prop"
          >
            Color:
          </span>
        </p>
      </div>
      <div
        class="actions"
      >
        <button
          class="btn"
          data-assignment="modalAddToCart"
          style="background-color: brown; color: bisque;"
        >
          Add to Cart
        </button>
        <div
          class="wrapStar"
        >
          <svg
            height="28"
            style="cursor: pointer;"
            viewBox="0 0 51 48"
            width="28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              Five Pointed Star
            </title>
            <path
              d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
              fill="bisque"
              stroke=""
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <div
    class="card"
    data-cards="true"
    data-testid="card"
    id="3"
  >
    <h2
      class="text title"
    />
    <img
      width="280"
    />
    <div
      class="cardContent"
    >
      <div>
        <p
          class="text"
        >
          <span
            class="prop"
          >
            Price:
          </span>
        </p>
        <p
          class="text"
        >
          <span
            class="prop"
          >
            Articul:
          </span>
          3
        </p>
        <p
          class="text"
        >
          <span
            class="prop"
          >
            Color:
          </span>
        </p>
      </div>
      <div
        class="actions"
      >
        <button
          class="btn"
          data-assignment="modalAddToCart"
          style="background-color: brown; color: bisque;"
        >
          Add to Cart
        </button>
        <div
          class="wrapStar"
        >
          <svg
            height="28"
            style="cursor: pointer;"
            viewBox="0 0 51 48"
            width="28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              Five Pointed Star
            </title>
            <path
              d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
              fill="bisque"
              stroke=""
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</section>
`;
